catroid/src/main/java/org/catrobat/catroid/ui/cloudmessaging/CustomFirebaseMessagingService.java

```diff
@@ -0,0 +1,115 @@
+import android.graphics.BitmapFactory;
+import android.media.RingtoneManager;
+import android.net.Uri;
+import android.support.v4.app.NotificationCompat;
+import android.util.Log;
+
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
+
+import org.catrobat.catroid.R;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Map;
+
+public class CustomFirebaseMessagingService extends FirebaseMessagingService {
+
+	public static final String TAG = CustomFirebaseMessagingService.class.getSimpleName();
+
+	@Override
+	public void onMessageReceived(RemoteMessage remoteMessage) {
+		RemoteMessage.Notification notification = remoteMessage.getNotification();
+
+		String title = notification.getTitle();
+		String message = notification.getBody();
+
+		Map<String, String> customData = remoteMessage.getData();
+
+		if (!customData.containsKey("image") || !customData.containsKey("link") || isStringNullOrEmpty(title) || isStringNullOrEmpty(message)) {
+			Log.e(TAG, "Cloud Notification Failed due to missing parameter(s)");
+			return;
+		}
+
+		String image = customData.get("image");
+		String link = customData.get("link");
+
+		Bitmap imageBitmap = getBitmap(image);
+
+		sendNotification(title, message, imageBitmap, link);
+	}
+
+	private void sendNotification(String title, String message, Bitmap image, String link) {
+		Log.d(TAG, "Title : " + title);
+		Log.d(TAG, "Body : " + message);
+		Log.d(TAG, "link : " + link);
+		Log.e(TAG, "image : " + image);
+
+		Uri webPage = Uri.parse(link);
+		Intent intent = new Intent(Intent.ACTION_VIEW, webPage);
+		PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, intent, 0);
+		Uri defaultSoundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
+
+		NotificationCompat.Builder notification = new NotificationCompat.Builder(this)
+				.setSmallIcon(R.drawable.ic_launcher)
+				.setLargeIcon(image)
+				.setContentTitle(title)
+				.setContentText(message)
+				.setAutoCancel(true)
+				.setSound(defaultSoundUri)
+				.setContentIntent(pendingIntent);
+
+		NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+		notificationManager.notify(1, notification.build());
+	}
+
+	private boolean isStringNullOrEmpty(String string) {
+		return string == null || string.isEmpty();
+	}
+
+	public Bitmap getBitmap(String link) {
+		try {
+			URL url = new URL(link);
+			HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+			connection.setDoInput(true);
+			connection.connect();
+			InputStream input = connection.getInputStream();
+			return BitmapFactory.decodeStream(input);
+		} catch (IOException e) {
+			Log.e(TAG, "Failed to download image. Using default icon: " + e);
+			return BitmapFactory.decodeResource(this.getResources(), R.drawable.ic_launcher);
+		}
```

Comment:
Even better! :+1:

btw Log.e has also a version with three parameters: `Log.e(TAG, "Failed to download image. Using default icon", e);`